src/vs/workbench/browser/parts/editor/tabsTitleControl.ts

```diff
@@ -337,8 +339,25 @@ export class TabsTitleControl extends TitleControl {
 				}
 			}
 
-			// Figure out scrolling direction
-			const nextEditor = this.group.getEditorByIndex(this.group.getIndexOfEditor(activeEditor) + (e.deltaX < 0 || e.deltaY < 0 /* scrolling up */ ? -1 : 1));
+			// Ignore event if the last one happened too recently.
+			// The restriction is relaxed according to the abs value of deltaX and deltaY
+			const time_threshold = 150;
+			if (Date.now() - this.lastMouseWheelEventTime < time_threshold - 2 * (Math.abs(e.deltaX) + Math.abs(e.deltaY))) {
+				return;
+			}
+			this.lastMouseWheelEventTime = Date.now();
+
+			// Figure out scrolling direction (ignore direction if the scrolling is too subtile)
+			const distance_threshold = 1.5;
+			let tabSwitchDirection = 0;
+			if (e.deltaX + e.deltaY < - distance_threshold) {
+				tabSwitchDirection = -1;
+			}
+			if (e.deltaX + e.deltaY > distance_threshold) {
+				tabSwitchDirection = 1;
+			}
+			const nextEditor = this.group.getEditorByIndex(this.group.getIndexOfEditor(activeEditor) + tabSwitchDirection);
```

Comment:
Could/should we return early from this method if we don't have a `tabSwitchDirection`? In other words, do we need the code to execute even if the same editor is going to be opened? I would prefer to ignore the work of opening an editor unless it really changes.

Reply:
You're right, we should return already if `tabSwitchDirection` is zero. I thought the `if` that checks if `nextEditor` is set was taking care of this, but it is not.